components/navigation/GardenStackNavigator.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+
+import MyGardenPage from '../pages/MyGarden'
+import PlantProfilePage from '../pages/PlantProfile'
+
+const GardenStack = createStackNavigator()
```

Comment:
Nesting, nesting, and more nesting! I like that you created separate components for the nested navigation components.